privmx-endpoint/src/main/cpp/modules/EventApi.cpp

```diff
@@ -106,47 +106,116 @@ Java_com_simplito_java_privmx_1endpoint_modules_event_EventApi_emitEvent(
+                    subscription_queries_c.push_back(ctx.jString2string((jstring) arrayElement));
+                }
+
+                auto subscription_ids_c = getEventApi(ctx, thiz)->subscribeFor(
+                        subscription_queries_c);
+
+                jobject javaArrayList = env->NewObject(arrayListCls, initMID);
+                for (auto &id_str : subscription_ids_c) {
+                    jstring java_id_str = ctx->NewStringUTF(id_str.c_str());
+                    env->CallBooleanMethod(javaArrayList, addToListMID, java_id_str);
+                }
+                return javaArrayList;
+            }
+    );
+
+    if (ctx->ExceptionCheck()) {
+        return nullptr;
+    }
+    return result;
 }
 
+
 extern "C"
 JNIEXPORT void JNICALL
-Java_com_simplito_java_privmx_1endpoint_modules_event_EventApi_unsubscribeFromCustomEvents(
+Java_com_simplito_java_privmx_1endpoint_modules_event_EventApi_unsubscribeFrom(
         JNIEnv *env,
         jobject thiz,
-        jstring context_id,
-        jstring channel_name
+        jobject subscription_ids
 ) {
     JniContextUtils ctx(env);
-
-    if (ctx.nullCheck(context_id, "Context ID") ||
-        ctx.nullCheck(channel_name, "Channel name")) {
+    if (ctx.nullCheck(subscription_ids, "Subscription IDs for CustomApi")) {
         return;
     }
 
-    ctx.callVoidEndpointApi([&ctx, &thiz, &context_id, &channel_name]() {
-        getEventApi(ctx, thiz)->unsubscribeFromCustomEvents(
-                ctx.jString2string(context_id),
-                ctx.jString2string(channel_name)
-        );
+    ctx.callVoidEndpointApi([&ctx, &env, &thiz, &subscription_ids]() {
+        auto subscription_ids_arr = ctx.jObject2jArray(subscription_ids);
+        auto subscription_ids_c = std::vector<std::string>();
+
+        int length = ctx->GetArrayLength(subscription_ids_arr);
+        for (int i = 0; i < length; i++) {
+            jobject arrayElement = ctx->GetObjectArrayElement(subscription_ids_arr, i);
+            subscription_ids_c.push_back(ctx.jString2string((jstring) arrayElement));
+        }
+
+        getEventApi(ctx, thiz)->unsubscribeFrom(subscription_ids_c);
     });
-}
\ No newline at end of file
+}
+
+
+extern "C"
+JNIEXPORT jstring JNICALL
+Java_com_simplito_java_privmx_1endpoint_modules_event_EventApi_buildSubscriptionQuery(
+        JNIEnv *env,
+        jobject thiz,
+        jstring channelName,
+        jlong selectorType,
+        jstring selectorId
+) {
+    JniContextUtils ctx(env);
+    if (ctx.nullCheck(channelName, "ChannelName") ||
+        ctx.nullCheck(selectorId, "SelectorID")) {
+        return nullptr;
+    }
+
+    jstring result = nullptr;
+    ctx.callResultEndpointApi<jstring>(
+            &result,
+            [&ctx, &env, &thiz, &channelName, &selectorType, &selectorId]() {
+                std::string c_channelName = ctx.jString2string(channelName);
+                auto c_selectorType = parseEventSelectorType(ctx, (long) selectorType);
```

Comment:
you can easy cast long value to enum like this
```suggestion
                auto c_selectorType = static_cast<event::EventSelectorType>(selectorType);
```